### PR TITLE
4.7

### PR DESCRIPTION
**Pull request by FrankXuuu**

### CODE REVIEW

File: src/pyasm/biz/task.py [hunk 1]
```diff
@@ -114,6 +114,7 @@
     "On Hold":  "#a96ccf",
     "Archived": "#DDDDDD",
     "Not Required": "#DDDDDD",
+    "Not Approved": "#e84a4d"
```
Comment (CeltonMcGrath) [inline src/pyasm/biz/task.py]: Is this still needed?

File: src/tactic/ui/tools/pipeline_wdg.py [hunk 2]
```diff
@@ -2038,17 +2040,21 @@ def get_display(self):
...
         
-        # If either the left process or right process do not exist,
-        # display empty pane.
-        if not left_process or not right_process:
-            info_wdg = DivWdg()
-            info_wdg.add_style("margin: 10px")
-            info_wdg.add("Save your workflow to edit connector properties.") 
-            top.add(info_wdg)
-            return top
+        # deprecated by sessional changes
```
Comment (CeltonMcGrath) [inline src/tactic/ui/tools/pipeline_wdg.py]: you can delete this

File: src/tactic/ui/tools/pipeline_wdg.py [hunk 3]
```diff
@@ -5978,6 +6121,8 @@ def get_buttons_wdg(self, show_gear):
 
             server = TacticServerStub.get();
             spt.app_busy.show("Saving project-specific pipeline ["+group_name+"]",null);
+
+            console.log(node_kwargs);
```
Comment (CeltonMcGrath) [inline src/tactic/ui/tools/pipeline_wdg.py:6149]: remove

Comment (FrankXuuu) [reply src/tactic/ui/tools/pipeline_wdg.py:6149]: this console log is really useful for debugging

**Pull request merged**